multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/SupportedParametersChecker.java

```diff
@@ -0,0 +1,114 @@
+                unsupportedAndNotReferencedParameters.add(parameter);
+            }
+        }
+        return unsupportedAndNotReferencedParameters;
+    }
+
+    private List<String> findUnsupportedParameters(DeploymentDescriptor descriptor) {
+        this.unknownParameters = new ArrayList<>();
+        descriptor.getModules()
+                  .forEach(this::checkModuleParameters);
+        descriptor.getResources()
+                  .forEach(this::checkResourceParameters);
+        checkParameters(getGlobalParameters(), descriptor.getParameters());
+        return unknownParameters;
+    }
+
+    private void checkModuleParameters(Module module) {
+        checkParameters(getModuleParameters(), module.getParameters());
+        getModuleHooks(module).forEach(this::checkHookParameters);
+        module.getRequiredDependencies()
+              .forEach(dependency -> checkParameters(getDependencyParameters(), dependency.getParameters()));
+        module.getProvidedDependencies()
+              .forEach(dependency -> checkParameters(getDependencyParameters(), dependency.getParameters()));
+
+    }
+
+    protected Set<String> getDependencyParameters() {
+        Set<String> dependencyParameters = new HashSet<>();
+        dependencyParameters.addAll(SupportedParameters.DEPENDENCY_PARAMETERS);
+        dependencyParameters.addAll(getModuleParameters());
+        dependencyParameters.addAll(getResourceParameters());
+        return dependencyParameters;
+    }
+
+    protected Set<String> getModuleParameters() {
+        return SupportedParameters.MODULE_PARAMETERS;
+    }
+
+    protected Set<String> getModuleHookParameters() {
+        return SupportedParameters.MODULE_HOOK_PARAMETERS;
+    }
+
+    protected Set<String> getResourceParameters() {
+        return SupportedParameters.RESOURCE_PARAMETERS;
+    }
+
+    protected Set<String> getGlobalParameters() {
+        return SupportedParameters.GLOBAL_PARAMETERS;
+    }
+
+    private void checkResourceParameters(Resource resource) {
+        checkParameters(getResourceParameters(), resource.getParameters());
+        checkRequiredDependencyParameters(getResourceRequiredDependencies(resource));
+    }
+
+    private void checkHookParameters(Hook hook) {
+        checkParameters(getModuleHookParameters(), hook.getParameters());
+        checkRequiredDependencyParameters(hook.getRequiredDependencies());
+    }
+
+    private void checkRequiredDependencyParameters(List<RequiredDependency> requiredDependencies) {
+        requiredDependencies.forEach(requiredDependency -> checkParameters(getDependencyParameters(), requiredDependency.getParameters()));
+    }
+
+    private void checkParameters(Set<String> supportedParameters, Map<String, Object> parameters) {
+        for (String parameterName : parameters.keySet()) {
+            if (!supportedParameters.contains(parameterName) && !getGlobalParameters().contains(parameterName)) {
```

Comment:
what is the point of allowing global parameters to pass the checks on all levels? e.g. APPS_START_TIMEOUT would pass these checks if defined at any level, even where it's not appropriate

Reply:
Copied some of the parameters to be also in supported module parameters and removed the additional global parameters check.